devito/propagator.py

```diff
@@ -282,12 +298,16 @@ def convert_equality_to_cgen(self, equality):
             s_lhs = ccode(self.time_substitutions(equality.lhs).xreplace(self._var_map))
             s_rhs = self.time_substitutions(equality.rhs).xreplace(self._var_map)
 
+            if s_lhs.find("temp") != -1:
+                s_lhs = "double" + " " + s_lhs
```

Comment:
I couldn't find a way to convert the dtype to the proper string easily so I left it this way while working on the actual CSE.